### ISSUE TITLE
Issues with KP selection?

### ISSUE THREAD

**Issue by edeutsch** [RTXteam/RTX#1905]
Hi, I'm doing a bit of testing of the automatic KP selection system and running into some unexpected results.

When I go to https://arax.ncats.io/devED
and run the example JSON query, I see things like this:
![image](https://user-images.githubusercontent.com/12707718/188167697-ef7a595c-01b6-4f78-af56-8eed4bd0ff3d.png)

First, it would be nice to augment the message:
KP does not have a TRAPI 1.3 endpoint
to
KP does not have a TRAPI 1.3 development endpoint

because for example infores:automat-foodb does appear to have some TRAPI 1.3 endpoints, but maybe not the one you're looking for. And so to head off reports of "hey, we do have a TRAPI 1.3 endpoint", it would be nice to list the maturity level desired and tried.

But second, it does appear that e.g. automat-foodb does have a development TRAPI 1.3 endpoint:
![image](https://user-images.githubusercontent.com/12707718/188168455-4a177aa6-4b1b-4f13-8812-249a6491c135.png)

And so I'm wondering why I'm seeing a notice that there is no TRAPI 1.3 endpoint.

What do you think?

**Comment by rcpeene** [comment]
The message _should_ read "KP does not have a TRAPI 1.3 development endpoint". It is an error that is displays the incorrect message. Looking into this strange behavior now.

**Comment by rcpeene** [comment]
I believe this error would happen when a KP has multiple Smart API registry entries of the same maturity where one entry was of a compatible version, but one wasn't. The implementation I wrote in smartapi.py would count the KP as 'excluded by version' because _one_ entry was excluded by version, even when other valid entries existed. This has been fixed and pushed to branch `issue1905`. A PR has been issued
